ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -215,10 +209,21 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
         const showTimeSeries = isSelected || isSelectAll;
 
         if (showTimeSeries) {
+          // Use timeChartData.length to ensure the data that is passed into the tooltip accounts for
+          // which legend items are selected.
+          const datasetIndex = timeChartData.length;
```

Comment:
Optional suggestion: It took me a minute to figure out what was going on here. It might be worth clarifying in the comment that this _must_ happen before the `timeChartData.push` below to avoid an off-by-one error. I could see someone accidentally breaking this while trying to refactor the component. It's also possible this is just "still getting back into work after a 2 week vacation" brain, so no big deal either way.

Reply:
Added the comment! By the way the diff here was out of date, this PR was originally going into branch for #1302 (which was merged into main this morning)

That PR has more info about why this change was needed, after the LineChart -> TimeChart refactor, clicking a legend item would cause tooltip to not render since it was looping through entire dataset instead of just the filtered data based on selected legend items